### ISSUE TITLE
consider putting details about errors in `models.VulnerabilityResults`

### ISSUE THREAD

**Issue by josieang** [google/osv-scanner#675]
This was discussed in: https://github.com/google/osv-scanner/pull/501#discussion_r1401646226

Return error codes don't seem to be the right way to report what combination of vulnerability, called vulnerabilities, license violations. It's simpler to just return 0 or 1, and put the details about what kinds of issues are present in the json or `models.VulnerabilityResults` output.

**Comment by oliverchang** [comment]
@cuixq would this be interesting for v2?

**Comment by cuixq** [comment]
yes, we would like to re-design error types in v2 https://github.com/google/osv-scanner/issues/636, and this is related.
